src/components/Profile.jsx

```diff
@@ -1,13 +1,150 @@
-import React, { useContext } from "react";
+import React, { useContext, useState } from "react";
 import { UserContext } from "../context/UserContext";
+import { Link, useParams } from "react-router-dom";
+import { userData } from "../UserData";
 
 const Profile = () => {
   const { name } = useContext(UserContext);
+  const [userList] = useState(userData);
+  const param = useParams();
+
+  const currentUser = param.name === name;
+
+  const userProfile = userList.find(
+    (thisUser) => thisUser?.name === param.name
+  );
+
+  const { id, firstName, lastName, designation, favoriteFood, hobbies } =
+    userProfile;
 
   return (
     <div className="container">
-      <h1 className="title title-1 profile-name">{name}'s profile</h1>
-      <p>Username: {name}</p>
+      <div key={id} className="profile">
+        <div className="profile-banner">
+          <img src="../5.jpg" alt="banner-pic" />
+        </div>
+
+        <div className="profile-basic">
+          <div className="profile-pic">
+            <img
+              src="https://static.vecteezy.com/system/resources/thumbnails/003/337/584/small/default-avatar-photo-placeholder-profile-icon-vector.jpg"
+              alt="profile-pic"
+            />
+          </div>
+
+          <div className="profile-intro">
+            <h1 className="title title-1 profile-name">
+              {firstName} {lastName}
+            </h1>
+            <div className="profile-designation">
+              Designation: {designation}
+            </div>
+          </div>
+
+          {currentUser && (
+            <Link className="button" to={`/edit-profile/${name}`}>
+              Edit Profile
+            </Link>
+          )}
+        </div>
+
+        <div className="profile-body">
+          <div className="profile-block">
+            <h4 className="title title-4">My Skills</h4>
+            <div className="skill-set">
+              <span>
+                <img
+                  src="https://upload.wikimedia.org/wikipedia/commons/thumb/3/38/HTML5_Badge.svg/2048px-HTML5_Badge.svg.png"
```

Comment:
without using this 3rd party source, let's keep it in the asset folder and import that from there